app/mobile-content/pages/shared/page-input/page-input.component.html

```diff
@@ -0,0 +1,161 @@
+<div class="page-form">
```

Comment:
That's a pretty good idea, actually. And would make adding new types easier. However it would also mean quite a bit of work. You don't want to review a 2000 LOC PR, do you? But keep the idea in mind as a possible improvement in the future.